### DISCUSSION TITLE
Alternate capture method

### DISCUSSION THREAD

**Discussion by yschimke** [PatilShreyas/Capturable#98] [General]
Could Capturable use this alpha API to capture the image instead of an AndroidView?

https://cs.android.com/androidx/platform/frameworks/support/+/f60a1149f00d6cc5972e9bd4057eb4a15505b468:compose/ui/ui-graphics/samples/src/main/java/androidx/compose/ui/graphics/samples/DrawScopeSample.kt;l=131

**Comment by PatilShreyas** [comment]
In Compose, `drawWithCache()` is used for drawing on the canvas of compose. i.e. you can draw text, lines, shapes, etc.
Refer: https://developer.android.com/jetpack/compose/graphics/draw/modifiers

**Reply by yschimke** [reply]
The API works with other forms, I thought it should be possible

> The retargeting feature can be used to capture a composable as a bitmap yes

https://slack-chats.kotlinlang.org/t/12321887/how-can-i-get-bitmap-from-composable-function#b2205524-3250-4505-9332-0fd2741aee8b

**Reply by PatilShreyas** [reply]
Oh great, thanks for the good reference. Will definitely take a look into this

**Comment by yschimke** [comment]
This is where I got to. https://gist.github.com/yschimke/2cbbf4e1fe2354beb10ce2e83216f7e3

Just using retargeting in drawWithCache to an alternative canvas.  Unfortunately it's still limited by the current screen. So not properly working. Not sure if this actually an improvement.

```
    val finalSize = DpSize(227.dp, 227.dp)
    SavedPicture(
        size = finalSize,
        onPicture = { lastPicture = SavedPicture(lastPicture.i + 1, it) },
        pictureTrigger = pictureTrigger,
    ) {
        Box(
            modifier = Modifier
                .fillMaxSize()
                .background(Color.DarkGray),
            contentAlignment = Alignment.Center
        ) {
            Text("Snapped\n" + LocalTime.now().withNano(0))
        }
    }
```


https://github.com/PatilShreyas/Capturable/assets/231923/d2fae5f3-1894-48db-9b29-a7a2180fe488

**Reply by PatilShreyas** [reply]
Current API has no support for that.

**Reply by yschimke** [reply]
I'll try with https://cs.android.com/androidx/platform/frameworks/support/+/fa28b73fb1bcba009b460aa0a46613ad2a41d9ed

**Comment by yschimke** [comment]
This would allow using outside a Composition

https://slack-chats.kotlinlang.org/t/18809650/compose-1-7-0-introduces-remembergraphicslayer-which-looks-l
